lib/common/asar.js

```diff
@@ -1,20 +1,311 @@
 (function () {
   const asar = process.binding('atom_common_asar')
-  const {Buffer} = require('buffer')
+  const { Buffer } = require('buffer')
   const childProcess = require('child_process')
   const path = require('path')
 
+  const LOCAL_DEBUG = false //
+  // integrities config
+  const integrities = {
+    // returns integrities as stringified json.
+    getIntegrity: function () {
+      if (typeof this.asarIntegrities === 'undefined') {
+        // default
+        this.asarIntegrities = {
+          checksums: {}
+        }
+
+        // get integrities from c/c++ source, if possible
+        const integrityRaw = asar.getIntegrity()
```

Comment:
why is this needed and where does it come from?

Reply:
Currently, it isn't needed. Except for macOS applications: the electron-builder puts its own checksums to the "Info.plist", which are prioritized to the manually added ones.